server2/src/main/java/se/salt/server2/domain/assignment/models/AssignmentCategory.java

```diff
@@ -12,4 +12,13 @@ public enum AssignmentCategory {
     public String getValue() {
         return value;
     }
+
+    public static AssignmentCategory fromString(String text) {
+        for (AssignmentCategory category : AssignmentCategory.values()) {
+            if (category.value.equalsIgnoreCase(text)) {
+                return category;
+            }
+        }
+        throw new IllegalArgumentException("No constant with text " + text + " found");
```

Comment:
IllegalArgument is fine but you could create an even more descriptive custom one like the one for account not found